src/logging.cc

```diff
@@ -1416,8 +1416,34 @@ bool LogCleaner::IsLogFromCurrentProject(
     }
   }
 
+  // cleaned_base_file contains .<severity>. at the end
+  // If we check it against filepath it will only allow files
+  // with that severity. As we want to delete old logs despite their severity,
+  // we need to replace <severity> in cleaned_base_file and find matching one. 
+
+  // Erase severity name from cleaned_base_filename
+  // Delete last dot to be able to use `find_last_of` function
+  cleaned_base_filename.pop_back();
+  int dot_pos = cleaned_base_filename.find_last_of('.');
+  cleaned_base_filename.erase(dot_pos + 1, std::string::npos);
```

Comment:
You do actually need to modify the string. Something like this should be sufficient:
```suggestion
  const string::size_type dot_pos = cleaned_base_filename.find_last_of('.', cleaned_base_filename.size() - 1);
```